t/unit/backends/test_base.py

```diff
@@ -534,17 +534,23 @@ def test_mark_as_revoked__chord(self):
         b.on_chord_part_return.assert_called_with(request, states.REVOKED, ANY)
 
     def test_chord_error_from_stack_raises(self):
+        class ExpectedException(Exception):
+            pass
+
         b = BaseBackend(app=self.app)
-        exc = KeyError()
         callback = Mock(name='callback')
         callback.options = {'link_error': []}
+        callback.keys.return_value = []
```

Comment:
This mock doesn't work on pypy3 :(
```
___________ test_BaseBackend_dict.test_chord_error_from_stack_raises ___________

self = <t.unit.backends.test_base.test_BaseBackend_dict object at 0x000000000aad6e20>

    def test_chord_error_from_stack_raises(self):
        class ExpectedException(Exception):
            pass
    
        b = BaseBackend(app=self.app)
        callback = Mock(name='callback')
        callback.options = {'link_error': []}
        callback.keys.return_value = []
        task = self.app.tasks[callback.task] = Mock()
        b.fail_from_current_stack = Mock()
        group = self.patching('celery.group')
        with patch.object(
            b, "_call_task_errbacks", side_effect=ExpectedException()
        ) as mock_call_errbacks:
>           b.chord_error_from_stack(callback, exc=ValueError())

t/unit/backends/test_base.py:550: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <celery.backends.base.BaseBackend object at 0x000000000a6552f0>
callback = <Mock name='callback' id='174412584'>, exc = ValueError()

    def chord_error_from_stack(self, callback, exc=None):
        app = self.app
        try:
            backend = app._tasks[callback.task].backend
        except KeyError:
            backend = self
        # We have to make a fake request since either the callback failed or
        # we're pretending it did since we don't have information about the
        # chord part(s) which failed. This request is constructed as a best
        # effort for new style errbacks and may be slightly misleading about
        # what really went wrong, but at least we call them!
        fake_request = Context({
            "id": callback.options.get("task_id"),
            "errbacks": callback.options.get("link_error", []),
            "delivery_info": dict(),
>           **callback
        })
E       TypeError: 'Mock' object is not a mapping
```

Reply:
`MagicMock`s appear to work okay for both Python3 and pypy3